src/Libraries/RevitNodes/Elements/TextNote.cs

```diff
@@ -135,18 +142,28 @@ private void Init(RVT.View view, RVT.XYZ origin, RVT.HorizontalTextAlignment ali
                     element.Text = text;
                 }
 
-                if (element.Location is LocationPoint)
+                if (element.Location == null)
                 {
-                    LocationPoint point = (LocationPoint)element.Location;
-
-                    if (!point.Point.Equals(origin))
+                    if (!element.Coord.Equals(origin))
                     {
-                        point.Point = origin;
+                        element.Coord = origin;
                     }
-
-                    if (point.Rotation != rotation.ToRadians())
+                }
+                else
+                {
+                    if (element.Location is LocationPoint)
                     {
-                        point.Rotate(Line.CreateUnbound(XYZ.Zero, XYZ.BasisZ), rotation.ToRadians());
+                        LocationPoint point = (LocationPoint)element.Location;
+
+                        if (!point.Point.Equals(origin))
+                        {
+                            point.Point = origin;
+                        }
+
+                        if (point.Rotation != rotation.ToRadians())
```

Comment:
same here - if this is a double comparison it should be done with some epsilon tolerance